homeassistant/components/script/config.py

```diff
@@ -47,19 +90,26 @@ async def _try_async_validate_config_item(hass, object_id, config, full_config=N
         async_log_exception(ex, DOMAIN, full_config or config, hass)
         return None
 
+    if isinstance(config, BlueprintInputs):
+        return config
+
     config = ScriptConfig(config)
     config.raw_config = raw_config
     return config
 
 
 async def async_validate_config(hass, config):
     """Validate config."""
-    if DOMAIN in config:
-        validated_config = {}
-        for object_id, cfg in config[DOMAIN].items():
+    scripts = {}
+    for _, p_config in config_per_platform(config, DOMAIN):
```

Comment:
This is what automations use at this point as well. I agree the helper name is weird, happy to address that in general in a follow-up PR (as it would affect around 30 uses in 16 files, unrelated to this PR).

The reason this is added is to make split config work and allows for splitting manual configuration from the `scripts.yaml` file which the UI uses to write to.

This mimicks the current behavior of automations, this would become more important as soon as a UI around these is created as well.